packages/cubejs-server-core/src/core/CompilerApi.js

```diff
@@ -187,6 +189,35 @@ export class CompilerApi {
     };
   }
 
+  async dataSources(options, orchestratorApi) {
```

Comment:
I would prefer if you'll keep `getCompilers` method in its original state, without changing its signature. Instead, you can partially duplicate its logic in the `dataSource` method. It is just a ~25 lines of code and I don't like this new `if` statement in the `getCompilers` method. You already added `dataSource` for your functionality, so lets split these two.

Reply:
done